manageiq-operator/pkg/helpers/miq-components/postgresql.go

```diff
@@ -175,63 +193,63 @@ func NewPostgresqlDeployment(cr *miqv1alpha1.ManageIQ) (*appsv1.Deployment, erro
+	err := addResourceReqs(cr.Spec.PostgresqlMemoryLimit, cr.Spec.PostgresqlMemoryRequest, cr.Spec.PostgresqlCpuRequest, cr.Spec.PostgresqlCpuLimit, &container)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	deploymentLabels := map[string]string{
-		"app": cr.Spec.AppName,
-	}
-	podLabels := map[string]string{
-		"name": "postgresql",
-		"app":  cr.Spec.AppName,
-	}
 	var repNum int32 = 1
 
-	deployment := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "postgresql",
-			Namespace: cr.ObjectMeta.Namespace,
-			Labels:    deploymentLabels,
+	spec := appsv1.DeploymentSpec{
+		Replicas: &repNum,
+		Selector: &metav1.LabelSelector{
+			MatchLabels: deploymentLabels,
 		},
-		Spec: appsv1.DeploymentSpec{
-			Strategy: appsv1.DeploymentStrategy{
-				Type: "Recreate",
-			},
-			Replicas: &repNum,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: podLabels,
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: deploymentLabels,
+				Name:   "postgresql",
 			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:   "postgresql",
-					Labels: podLabels,
-				},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{container},
-					Volumes: []corev1.Volume{
-						corev1.Volume{
-							Name: "miq-pgdb-volume",
-							VolumeSource: corev1.VolumeSource{
-								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-									ClaimName: "postgresql",
-								},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{container},
+				Volumes: []corev1.Volume{
+					corev1.Volume{
+						Name: "miq-pgdb-volume",
+						VolumeSource: corev1.VolumeSource{
+							PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+								ClaimName: "postgresql",
 							},
 						},
-						corev1.Volume{
-							Name: "miq-pg-configs",
-							VolumeSource: corev1.VolumeSource{
-								ConfigMap: &corev1.ConfigMapVolumeSource{
-									LocalObjectReference: corev1.LocalObjectReference{Name: "postgresql-configs"},
-								},
+					},
+					corev1.Volume{
+						Name: "miq-pg-configs",
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{Name: "postgresql-configs"},
 							},
 						},
 					},
 				},
 			},
 		},
 	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "postgresql",
+			Namespace: cr.ObjectMeta.Namespace,
+		},
+	}
 
-	return deployment, nil
+	f := func() error {
+		if err := controllerutil.SetControllerReference(cr, deployment, scheme); err != nil {
+			return err
+		}
+		if deployment.ObjectMeta.Labels == nil {
+			deployment.ObjectMeta.Labels = make(map[string]string)
+		}
+		deployment.ObjectMeta.Labels["app"] = cr.Spec.AppName
+		deployment.Spec = spec
```

Comment:
I think this will also run into https://github.com/ManageIQ/manageiq-pods/issues/550 as it is currently, but we can fix that separately.